src/main/java/hu/elte/eserial/exception/EserialMapperMismatchException.java

```diff
@@ -0,0 +1,11 @@
+package hu.elte.eserial.exception;
+
+/**
+ * Thrown when serialization is interrupted due to a mismatched mapper.
+ */
+public class EserialMapperMismatchException extends EserialException {
+
+    public EserialMapperMismatchException(Class<?> actual) {
```

Comment:
The expected class or some more context would be useful.

Reply:
What should be the value of the expected class in case of a CompoundMapper?

Reply:
There could be two constructors, one with two `Class`es as inputs, and one with the actual `Class` and a `String` for the expected "Class". And this can be called with "compound object". It should be enough info for the user that an Integer is not a compound object.